src/server/util/authenticationHelpers.js

```diff
@@ -1,24 +1,79 @@
 const { Pool } = require('pg');
 const { sql } = require('./sqlHelpers');
-const { saltHashPassword } = require('./pwHash');
+const { saltHashPassword, validateHashPassword } = require('./pwHash');
 
 const pool = new Pool();
 
-const getUser = async function(req, res, next) {
-    const { username, email } = req.body;
+const getUserInDatabase = async function(username, email) {
     const client = await pool.connect();
-
     const result = await client.query(sql`SELECT * FROM users WHERE email = ${email} OR username = ${username};`);
     client.release();
+    return result.rows;
+}
+
+const userExistsInDatabase = async function(username, email) {
+    const rows = await getUserInDatabase(username, email);
+    let user, exists = rows.length > 0;
+    if (exists) {
+        user = rows[0];
+        if (username) exists = exists && user.username === username;
+        if (email) exists = exists && user.email === email;
```

Comment:
I'm not sure I follow, why are we resetting the value of `exists` in these statements?

Reply:
If we do want to keep this check/update for `exist`, you can write the if-statement a little simpler since it's already in an if-statement that checks the truthiness of `exists`. For example:

`if (username) exists = user.username === username;`

Reply:
I had this because I wanted to make sure that if the POST request sent an email that matched a user but a username that didn't match the user, then we can say that the requested resource isn't found.